cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/ConcurrentStreamWriter.java

```diff
@@ -111,6 +111,11 @@
     this.createLock = new ReentrantLock();
   }
 
+  public void close(Id.Stream streamId) throws IOException, NotFoundException {
+    EventQueue eventQueue = getEventQueue(streamId);
```

Comment:
There is already - which I overlooked before - a closeEventQueue() inside the StreamFileFactory inner class. I'll move out of the inner class and call it from here. The lock is a good point, thanks.